src/server/test/web/obviusTest.js

```diff
@@ -58,5 +62,210 @@ mocha.describe('Obvius API', () => {
+			});
+			mocha.it('should return accurate config file manifests', async () => {
+				const password = 'password';
+				const hashedPassword = await bcrypt.hash(password, 10);
+				const obviusUser = new User(undefined, 'obvius@example.com', hashedPassword, User.role.OBVIUS);
+				await obviusUser.insert(conn);
+				obviusUser.password = password;
+				const uploadRequestMode = 'CONFIGFILEUPLOAD';
+				const manifestRequestMode = 'CONFIGFILEMANIFEST';
+
+				// Adapted from ../obvius/README.md
+				const configFilePath = 'src/server/test/web/obvius/mb-001.ini.gz';
+				const upload = await chai.request(app)
+					.post('/api/obvius')
+					.field('username', obviusUser.username)
+					.field('password', obviusUser.password)
+					.field('mode', uploadRequestMode)
+					.field('serialnumber', 'mb-001')
+					.field('modbusdevice', '1234')
+					.attach('files', configFilePath);
+
+				const res = await chai.request(app)
+					.post('/api/obvius')
+					.field('username', obviusUser.username)
+					.field('password', obviusUser.password)
+					.field('mode', manifestRequestMode);
+
+				//logfile upload should respond with 200, success
+				expect(upload).to.have.status(200);
+
+				//logfile request should respond with 200, success
+				expect(res).to.have.status(200);
+
+				//get "all" config files to compare to response
+				const allConfigfiles = await Configfile.getAll(conn);
+				let response = '';
+				for (f of allConfigfiles) {
+					response += `CONFIGFILE,${f.makeFilename()},${f.hash},${f.created.format('YYYY-MM-DD hh:mm:ss')}`;
+				}
+
+				//the third line of the response should be the config file
+				expect(res.text.split("\n")[2]).equals(response);
+
+				//config file uploads should create accurate meter objects
+				const allMeters = await Meter.getAll(conn);
+
+				//mb-001.ini should make 28 meters
+				expect(allMeters.length).to.equal(28);
+
+				//these arrays should vary for different submeters
+				const meterNames = [];
+				const meterIDs = [];
+
+				//flags for meter fields (.type, .displayable, .enabled)
+				const allMetersAreObvius = true;
+				const allMetersAreNotDisplayable = true;
+				const allMetersAreNotEnabled = true;
+
+				//expected names and ids
+				const expMeterNames = [
+					'mb-001.0', 'mb-001.1', 'mb-001.2', 'mb-001.3', 'mb-001.4', 'mb-001.5', 'mb-001.6', 'mb-001.7',
+					'mb-001.8', 'mb-001.9', 'mb-001.10', 'mb-001.11', 'mb-001.12', 'mb-001.13', 'mb-001.14', 'mb-001.15',
+					'mb-001.16', 'mb-001.17', 'mb-001.18', 'mb-001.19', 'mb-001.20', 'mb-001.21', 'mb-001.22', 'mb-001.23',
+					'mb-001.24', 'mb-001.25', 'mb-001.26', 'mb-001.27'
+				];
+
+				const expMeterIDs = [
+					1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28
+				];
+				for (const meter of allMeters) {
+					//populate arrays with varying values in ascending order
+					let currentName = meter.name + "";
```

Comment:
Is the ``+ ""`` to force a string? Maybe use string directly to do this. If it stays then use single quotes since OED prefers that. This assumes it was not already a string (which I would have hoped it is but maybe not in JS).

Reply:
That was the idea, but as you said it isn't actually required and has been removed.